src/backend/drm/compositor/mod.rs

```diff
@@ -1640,6 +1640,22 @@ where
             .overlay
             .sort_by_key(|p| std::cmp::Reverse(p.zpos.unwrap_or_default()));
 
+        let driver = surface.get_driver().map_err(|err| {
+            FrameError::DrmError(DrmError::Access(AccessError {
+                errmsg: "Failed to query drm driver",
+                dev: surface.dev_path(),
+                source: err,
+            }))
+        })?;
+        // `IN_FENCE_FD` makes commit fail on Nvidia driver
+        // https://github.com/NVIDIA/open-gpu-kernel-modules/issues/622
+        let is_nvidia = driver.name().to_string_lossy().to_lowercase().contains("nvidia")
+            || driver
+                .description()
+                .to_string_lossy()
+                .to_lowercase()
+                .contains("nvidia");
```

Comment:
Does the open-driver also set `nvidia` or is it possibly `nvidia-open` and we need to filter for that as well?

Reply:
https://github.com/NVIDIA/open-gpu-kernel-modules/blob/083cd9cf17ab95cd6f9fb50a5349c21eaa2f7d4b/kernel-open/nvidia-drm/nvidia-drm-drv.c#L1666-L1669

Looks like NVIDIA's open kernel modules just advertise the driver as `"nvidia-drm"`, `"NVIDIA DRM driver"`, `"20160202"`. Like the closed source kernel driver does.